### PR TITLE
feat(shared): implement getParticipantActorType function and related tests

### PR DESCRIPTION
**Pull request by AMarcosCastelo**
### Summary

- Added the `getParticipantActorType` function to determine the actor type (`WASTE_GENERATOR`, `PROCESSOR`, `RECYCLER`) based on document events.
- Enhanced document getters tests to cover various scenarios for participant actor type retrieval.
- Introduced `MassIDDocumentActorType` enum to categorize actor types.
- Updated existing document getters to utilize new constants for event names.

---

- [x] **I, the PR author, declare that this PR works as expected and does not break any service. I also declare that I gave my best to apply all of the best practices and that I'm leaving the code better than I found it.**

## Summary by CodeRabbit

- **New Features**
  - Introduced a new function to determine participant roles based on document events, enhancing event processing accuracy.
  - Added new constants to support the expanded functionality related to document events.

- **Tests**
  - Expanded test coverage for the new functionality, validating various scenarios and edge cases in event handling.

### CODE REVIEW

**Review by cris-santos — DISMISSED** [review]

File: libs/shared/methodologies/bold/types/src/enum.types.ts [hunk 1]
```diff
@@ -45,6 +45,13 @@ export enum DocumentSubtype {
   WOOD_AND_WOOD_PRODUCTS = 'Wood and Wood Products',
 }
 
+export enum MassIDDocumentActorType {
```
Comment (cris-santos) [inline libs/shared/methodologies/bold/types/src/enum.types.ts]:
```suggestion
export enum MassIdDocumentActorType {
```
